source/MaterialXTest/MaterialXGenShader/GenShaderUtil.cpp

```diff
@@ -761,13 +762,29 @@ void ShaderGeneratorTester::validate(const mx::GenOptions& generateOptions, cons
             mx::ShaderRefPtr shaderRef = targetElement->asA<mx::ShaderRef>();
             mx::NodePtr outputNode = targetElement->asA<mx::Node>();
             mx::NodeDefPtr nodeDef = nullptr;
+            bool containsNodedefToSkip = false;
             if (output)
             {
                 outputNode = output->getConnectedNode();
                 // Handle connected upstream material nodes later on.
                 if (outputNode->getType() != mx::MATERIAL_TYPE_STRING)
                 {
                     nodeDef = outputNode->getNodeDef();
+
+                    // We sometimes have complex graph where the node we want to skip is upstream:
```

Comment:
The geompropvalue graphs have an extra test node downstream when the final value is not GLSL compatible. We need to traverse to find the unimplementable nodes.